contracts/EthStorageContract.sol

```diff
@@ -130,9 +159,13 @@ contract EthStorageContract is StorageContract, Decoder {
 
     // Write a large value to KV store.  If the KV pair exists, overrides it.  Otherwise, will append the KV to the KV array.
     function putBlob(bytes32 key, uint256 blobIdx, uint256 length) public payable virtual {
-        bytes32 dataHash = bytes32(0);
+        bytes32 dataHash = BlobHashGetter.getBlobHash(hashGetter, blobIdx);
         uint256 kvIdx = _putInternal(key, dataHash, length);
 
         emit PutBlob(kvIdx, length, dataHash);
     }
+
+    function getHashByKvIdx(uint256 kvIdx) public view returns (bytes32) {
```

Comment:
This function can be replaced by the getKvMetas() in its parent contract DecentralizedKV